parser_library/src/analyzer.cpp

```diff
@@ -67,13 +69,36 @@ std::unique_ptr<processing::preprocessor> analyzer_options::get_preprocessor(
 
     struct combined_preprocessor : processing::preprocessor
     {
+        mutable std::vector<std::unique_ptr<semantics::preprocessor_statement_si>> m_statements;
+
         std::vector<std::unique_ptr<processing::preprocessor>> pp;
         document generate_replacement(document doc) override
         {
+            m_statements.clear();
+
             for (const auto& p : pp)
                 doc = p->generate_replacement(std::move(doc));
             return doc;
         }
+
+        void collect_statements(
+            std::vector<std::unique_ptr<semantics::preprocessor_statement_si>>& statement_collector) override
+        {
+            statement_collector.insert(statement_collector.end(),
+                std::make_move_iterator(m_statements.begin()),
+                std::make_move_iterator(m_statements.end()));
+        }
+
+        const std::vector<std::unique_ptr<semantics::preprocessor_statement_si>>& get_statements() const override
+        {
+            for (const auto& p : pp)
+            {
+                p->collect_statements(m_statements);
+            }
+
+            return m_statements;
+        }
+
```

Comment:
seems like all the implementations are actually the same? Maybe it should not be a part of the virtual interface, but just simple base members?

Reply:
Yeah, I wanted to potentially do this in one of the next PRs but I can change it now as well as this approach is suitable for Endevor and CICS (not sure about DB2 yet but I guess it's going to be fine there as well).